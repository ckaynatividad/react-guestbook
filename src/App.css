.App {
  text-align: center;
  font-family: Georgia, 'Times New Roman', Times, serif;
}

span {
  font-weight: bold;
}

header {
  margin-top: 5%;
  margin-left: 2%;
  position: fixed;

  font-style: italic;
}

.mode-button {
  margin-left: -50%;
  width: 99px;
}
