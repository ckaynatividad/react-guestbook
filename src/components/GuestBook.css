.yeah {
  width: 500px;
}

#entry-box {
  background-color: red;
}
