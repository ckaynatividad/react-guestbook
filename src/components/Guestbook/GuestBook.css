.form {
  display: flex;
  flex-direction: column;
  line-height: 30px;
  width: 200px;
  position: fixed;
  margin-left: 200px;
  margin-top: 150px;
}

.yeah {
  width: 200px;
  height: 50px;
}
