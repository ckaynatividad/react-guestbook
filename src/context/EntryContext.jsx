import { createContext, useContext, useState } from 'react';

const EntryContext = createContext();

const EntryProvider = ({ children }) => {
  const [entries, setEntries] = useState([]);

  return (
    <EntryContext.Provider value={{ entries, setEntries }}>
      {children}
    </EntryContext.Provider>
  );
};

const useEntries = () => {
  const context = useContext(EntryContext);

  if (context === undefined) {
    throw new Error('must be defined with entryprovider');
  }
  return context;
};

export { EntryProvider, useEntries };
