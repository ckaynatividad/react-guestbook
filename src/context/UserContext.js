import { createContext, useContext, useMemo, useState } from 'react';

export const UserContext = createContext();

const UserProvider = ({ children }) => {
  const [user, setUser] = useState(null);
  console.log(process.env.AUTH_EMAIL);
  const login = (email, password) => {
    const loginSuccessful =
      email === process.env.AUTH_EMAIL &&
      password === process.env.AUTH_PASSWORD;
    if (loginSuccessful) setUser({ email });
    return loginSuccessful;
  };

  const logout = (callback) => {
    setUser(null);
    callback();
  };

  const value = { user, setUser, login, logout };

  return <UserContext.Provider value={value}>{children}</UserContext.Provider>;
};
const useUser = () => {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error('must be used within userprovider');
  }
  return context;
};

export { UserProvider, useUser };
