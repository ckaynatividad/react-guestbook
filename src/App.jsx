import { GuestbookView } from './views/Guestbook/GuestbookView';
import './App.css';
import { Header } from './components/Guestbook/Header';
import { fetchApi } from './services/api.js';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import Home from './views/Home/Home';
import { ThemeProvider } from 'styled-components';
import { GlobalStyles } from './components/Theme/globalStyles';
import { darkTheme, lightTheme } from './components/Theme/Themes';
import { useState } from 'react';

export default function App() {
  const [theme, setTheme] = useState('light');
  const themeToggler = () => {
    theme === 'light' ? setTheme('dark') : setTheme('light');
  };
  return (
    <ThemeProvider theme={theme === 'light' ? lightTheme : darkTheme}>
      <GlobalStyles />
      <div className="App">
        <BrowserRouter>
          <Switch>
            <Route exact path="/">
              <Home />
            </Route>
            <Route path="/guestbook">
              <Header />
              <GuestbookView />
            </Route>
          </Switch>
        </BrowserRouter>
      </div>
    </ThemeProvider>
  );
}
