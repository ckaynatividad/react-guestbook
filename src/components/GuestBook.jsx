import React, { useState } from 'react';
import { useEntries } from '../context/EntryContext';
import { useUser } from '../context/UserContext';

export default function GuestBook() {
  const [name, setName] = useState('');
  const [entry, setEntry] = useState('');
  const { user, setUser } = useUser();
  const [entries, setEntries] = useEntries();

  function updateList() {
    if (!entry) return;
    setUser(name);
    setEntries([...entries, { name, message: entry }]);
    setEntry('');
  }

  const handleSubmit = (e) => {
    e.preventDefault();
    updateList();
  };

  const guestForm = (
    <div>
      <label for="guestName">Guest Name</label>
      <input
        id="guestName"
        type="text"
        placeholder="Enter Name"
        value={name}
        onChange={(e) => setName(e.target.value)}
      />
    </div>
  );

  const displayMsg = user
    ? `thanks ${user}, you can look at other entries if there are any`
    : 'please sign';

  return (
    <div>
      <p>{displayMsg}</p>
      <form onSubmit={handleSubmit}>
        {user ? null : guestForm}
        <label for="entry">Guest Entry</label>

        <textarea
          id="entry"
          value={entry}
          placeholder="Type here"
          onChange={(e) => setEntry(e.target.value)}
        />
        <button type="submit">Submit</button>
        {user && (
          <button
            type="button"
            onClick={() => {
              setUser('');
              setName('');
            }}
          >
            Not {user}?
          </button>
        )}
      </form>
    </div>
  );
}
