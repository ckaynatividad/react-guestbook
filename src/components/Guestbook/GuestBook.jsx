import React, { useState } from 'react';
import { useEntries } from '../../context/EntryContext';
import { useUser } from '../../context/UserContext';
import './GuestBook.css';

export default function GuestBook() {
  const [name, setName] = useState('');
  const [entry, setEntry] = useState('');
  const { user, setUser } = useUser();
  const { entries, setEntries } = useEntries();

  function updateList() {
    if (!entry) return;
    setUser(name);
    setEntries([...entries, { name, message: entry }]);
    setEntry('');
  }

  const handleSubmit = (e) => {
    e.preventDefault();
    updateList();
  };

  const guestForm = (
    <div>
      <label>
        Name: &nbsp;
        <input
          id="guestName"
          type="text"
          value={name}
          onChange={(e) => setName(e.target.value)}
        />
      </label>
    </div>
  );

  return (
    <div className="form">
      <form onSubmit={handleSubmit}>
        {user ? <p>Your entry has been submitted.</p> : guestForm}
        {!user && (
          <label>
            Your Entry:&nbsp;
            <input
              className="yeah"
              type="text"
              id="entry-box"
              value={entry}
              onChange={(e) => setEntry(e.target.value)}
            />
          </label>
        )}
        <p>
          {!user && (
            <button className="submit" type="submit">
              Submit
            </button>
          )}
          {user && (
            <button
              type="button"
              onClick={() => {
                setUser('');
                setName('');
              }}
            >
              Not {user}?
            </button>
          )}
        </p>
      </form>
    </div>
  );
}
