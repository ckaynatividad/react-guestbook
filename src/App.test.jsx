import { render, screen } from '@testing-library/react';
import App from './App';
import GuestBook from './components/GuestBook';
import { EntryProvider } from './context/EntryContext';
import { UserProvider } from './context/UserContext';
import userEvent from '@testing-library/user-event';

test('renders guestbooks in app', () => {
  render(
    <UserProvider>
      <App />
    </UserProvider>
  );

  const header = screen.getByText(/write in guestbook to sign in/i);
  expect(header).toBeInTheDocument();

  const input = screen.getByLabelText(/Guest Entry/i);
  expect(input).toBeInTheDocument();
});

test('renders list', () => {
  render(
    <UserProvider>
      <App />
    </UserProvider>
  );
  const name = screen.getByLabelText(/Guest Name/i);
  userEvent.type(name, 'meow');

  const entry = screen.getByLabelText(/Guest Entry/i);
  userEvent.type(entry, 'meows entry');

  const button = screen.getByRole('button');
  userEvent.click(button);

  const msg = screen.getByRole('list');
  expect(msg).toBeInTheDocument();
});

test('guestbook works', () => {
  const container = render(
    <UserProvider>
      <EntryProvider>
        <GuestBook name="meow" entry="meows entry" />
      </EntryProvider>
    </UserProvider>
  );
  expect(container).toMatchSnapshot();
});
