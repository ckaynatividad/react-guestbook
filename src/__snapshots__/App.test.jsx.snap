// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`guestbook works 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <form>
          <div>
            <label>
              Guest Name:
              <input
                id="guestName"
                placeholder="Enter Name"
                type="text"
                value=""
              />
            </label>
          </div>
          <label>
            Guest Entry:
            <input
              class="yeah"
              id="entry-box"
              placeholder="Type here"
              type="text"
              value=""
            />
          </label>
          <button
            type="submit"
          >
            Submit
          </button>
          
        </form>
        <p>
          please sign guest book
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <form>
        <div>
          <label>
            Guest Name:
            <input
              id="guestName"
              placeholder="Enter Name"
              type="text"
              value=""
            />
          </label>
        </div>
        <label>
          Guest Entry:
          <input
            class="yeah"
            id="entry-box"
            placeholder="Type here"
            type="text"
            value=""
          />
        </label>
        <button
          type="submit"
        >
          Submit
        </button>
        
      </form>
      <p>
        please sign guest book
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
